Services/Services.API/Controllers/DoctorsController.cs

```diff
@@ -0,0 +1,147 @@
+
+        public DoctorsController(IDoctorService doctorService, ILogger<DoctorsController> logger, IMapper mapper,
+            IValidator<CreateDoctorDto> createDoctorDtoValidator,
+            IValidator<UpdateDoctorDto> updateDoctorDtoValidator)
+        {
+            _doctorService = doctorService;
+            _logger = logger;
+            _mapper = mapper;
+            _createDoctorDtoValidator = createDoctorDtoValidator;
+            _updateDoctorDtoValidator = updateDoctorDtoValidator;
+        }
+
+        /// <summary>
+        /// Gets the list of all doctors.
+        /// </summary>
+        /// <remarks>
+        /// Sample request:
+        ///
+        ///     GET /api/Doctors
+        ///
+        /// </remarks>
+        /// <returns>Returns a list of doctors objects.</returns>
+        /// <response code="200">Returns the list of doctors</response>
+        /// <response code="500">If there was an internal server error</response>
+        [HttpGet]
+        public async Task<IEnumerable<DoctorDto>> Get(CancellationToken cancellationToken)
+        {
+            var doctors = await _doctorService.GetAll(cancellationToken);
+            _logger.LogInformation("Requested doctors list");
+
+            var doctorDtos = _mapper.Map<IEnumerable<DoctorDto>>(doctors);
+            return doctorDtos;
+        }
+
+        /// <summary>
+        /// Gets a doctor by ID.
+        /// </summary>
+        /// <param name="id">The ID of the doctor to retrieve.</param>
+        /// <returns>Returns the doctor object.</returns>
+        /// <response code="200">If the doctor is found</response>
+        /// <response code="400">If validation errors occured</response>
+        /// <response code="404">If the doctor is not found</response>
+        /// <response code="500">If there was an internal server error</response>
+        [HttpGet("{id}")]
+        public async Task<DoctorDto> Get(string id, CancellationToken cancellationToken)
+        {
+            var doctor = await _doctorService.Get(Guid.Parse(id), cancellationToken);
+            _logger.LogInformation("Requested doctor with id {id}", id);
+            return _mapper.Map<DoctorDto>(doctor);
+        }
+
+        /// <summary>
+        /// Create a doctor by ID.
+        /// </summary>
+        /// <param name="createDoctorDto">The doctor object fields containing details.</param>
+        /// <response code="200">If the doctor is created</response>
+        /// <response code="400">If validation errors occured</response>
+        /// <response code="500">If there was an internal server error</response>
+        [HttpPost]
+        public async Task Post([FromBody] CreateDoctorDto createDoctorDto, CancellationToken cancellationToken)
+        {
+            var doctorValidation = await _createDoctorDtoValidator.ValidateAsync(createDoctorDto, cancellationToken);
+            if (!doctorValidation.IsValid)
```

Comment:
This is duplicating code, move it somewhere

Reply:
It would be even better if you used ValidateAndThrowAsync

Reply:
Fixed